complete some rendering TODOs

Rexicon226 opened the pull request:
The codebase is quite interesting, and I found it very fun to look through.

Follow-up question, should we end the global decls and error sets with a semi-colon?
Currently they render without one.
![image](https://github.com/andrewrk/autodoc/assets/87927264/daaddcfa-11a3-4d31-8786-594352c9b157)

Rexicon226 committed:
add `error_set` rendering

Rexicon226 committed:
add `global_const` rendering

Rexicon226 commented on lib/main.js:
```diff
@@ -307,6 +307,34 @@
+
+      const members = namespaceMembers(decl_index, false);
+      if (members.length !== 0) {
+        resizeDomList(domListErrSets, members.length, '<li><a href="#"></a></li>');
```
I think this is the right html signature? Please correct me.

alichraghi replied:
`<a>` in `<li>`? yes that's correct

Rexicon226 committed:
add docstring for global const

andrewrk commented:
[here's a page to compare it to](https://ziglang.org/documentation/master/std/#A;std:fs.MAX_PATH_BYTES)

Rexicon226 commented:
> [here's a page to compare it to](https://ziglang.org/documentation/master/std/#A;std:fs.MAX_PATH_BYTES)

I see. So we should insert the docstring in the view as well. Can do now!

Rexicon226 commented:
@andrewrk Should the docstring be above or below the source? I personally think it should be above, but your choice.

andrewrk commented:
Exercise taste! Show me your vision :)

Rexicon226 commented:
My computer clock once again decided that it is yesterday, but I pushed!
![image](https://github.com/andrewrk/autodoc/assets/87927264/821eb833-2f6d-4b79-b593-6019104b75c5)

andrewrk commented:
Can I see a screenshot of the error set page?

[comparison](https://ziglang.org/documentation/0.9.0/std/#std;event.Loop.Request.Msg.OpenAt.Error)

[note that current autodocs is regressed from this](https://ziglang.org/documentation/master/std/#A;std:posix.OpenError)

Rexicon226 commented:
It's currently in the middle of those two 😄 
I can instead of just rendering source, parse them like in the 0.9 docs. Give me a like 20 minutes to get home from school.
**this was all misguided**

Rexicon226 committed:
add error view properly this time

add error docstrings

Rexicon226 commented:
*Now* error sets work :)
Docstrings are a TODO, as I need to figure out how to best way to do them. Expect a follow up tomorrow. I think I've gotten over the Dunning-Kruger hill now, but time will tell.

![image](https://github.com/andrewrk/autodoc/assets/87927264/02627ced-fdd3-411a-8219-a85b2c83ce46)

Rexicon226 commented:
I've actually realized docstrings aren't that hard. Will push in a bit.

andrewrk commented on lib/index.html:
```diff
@@ -364,6 +364,11 @@ <h2>Error Sets</h2>
       <ul id="listErrSets">
       </ul>
     </div>
+    <div id="sectErrors" class="hidden">
+      <h2>Errors</h2>
+      <ul id="listErrors">
+      </ul>
+    </div>
```
The section is already above, line 313

Rexicon226 replied:
See it!

andrewrk commented:
previous error set implementation: https://github.com/andrewrk/autodoc/blob/a56d2c180084a065b12660d63cf0e88d502800d4/lib/main.js#L528-L560

Rexicon226 commented:
Sorry @andrewrk, I tabbed back to this GitHub page only once I had pushed. I didn't see you reviewing. I just cleaned it up a bit and added doc strings. I'm terrible at naming so please suggest fixes there as well :)

andrewrk commented:
can you limit the PR to one enhancement? let's start with the const view, yeah?

Rexicon226 commented:
Gotcha. Apologies for being all over the place today :) I feel this PR has gotten very noisy and it's all my fault. Would you like me to close this and open two new ones for each feature? (global const and error set)?
